### PR TITLE
Add Go wrapper support for different disk formats and drivers.

### PR DESCRIPTION
**Pull request by dlorenc**
This PR adds simple support for pre-configured disks of different types. Specifically, I'd like to be able to use the go wrapper to create VMs with attached qcow2 and sparsebundle disk types.

Apologies in advance for sending this PR without discussion on an issue first. I'm more than happy to refactor this or approach it in some other way.

Signed-off-by: dlorenc <lorenc.d@gmail.com>

### CODE REVIEW

File: go/hyperkit.go [hunk 1]
```diff
@@ -428,7 +430,18 @@ func (h *HyperKit) buildArgs(cmdline string) {
 	}
 
 	for _, p := range h.Disks {
-		a = append(a, "-s", fmt.Sprintf("%d:0,virtio-blk,%s", nextSlot, p.Path))
+		// Default the driver to virtio-blk
+		driver := "virtio-blk"
+		if p.Driver == "" {
```
Comment (ijc) [inline go/hyperkit.go]: Shouldn't this be a `!=` ?

File: go/hyperkit.go [hunk 2]
```diff
@@ -428,7 +430,18 @@ func (h *HyperKit) buildArgs(cmdline string) {
...
+			driver = p.Driver
+		}
+		arg := fmt.Sprintf("%d:0,%s,%s", nextSlot, driver, p.Path)
+
+		// Add on a format instruction if specified.
+		if p.Format != "" {
+			arg += fmt.Sprintf(",format=%s", p.Format)
```
Comment (ijc) [inline go/hyperkit.go]: I think this can be just `arg += ",format=" + p.Format` and therefore avoid the `fmt.Sprintf`.

File: go/hyperkit.go (hunk 1 above)
Comment (dlorenc) [inline go/hyperkit.go]: D'oh. That's what I get for manually copying the changes back out of my vendor directory...

File: go/hyperkit.go (hunk 2 above)
Comment (dlorenc) [inline go/hyperkit.go]: Done.

**Comment by dlorenc** [comment]
Thanks for the review! Any idea if the cirlcleci failure is related to my change?

**Comment by ijc** [comment]
```
clang: error: no such file or directory: '/Users/distiller/.opam/system/lib/io-page/io_page_unix.a'
clang: error: no such file or directory: '/Users/distiller/.opam/system/lib/io-page/libio_page_unix_stubs.a'
make: *** [build/hyperkit.sym] Error 1
```
Looks like one for @djs55.

@dlorenc if you could squash while he takes a look that'd be great. Your changes LGTM by themselves.

**Comment by djs55** [comment]
@dlorenc I think you're missing a build fix in 4afda334 -- rebasing to latest master should hopefully fix the CI.

**Comment by dlorenc** [comment]
@djs55 and @ijc thanks! squashed and rebased.

**Comment by dlorenc** [comment]
Looks like the master rebase did fix CI. Thanks!

**Pull request merged**